examples/input.rs

```diff
@@ -0,0 +1,316 @@
+                input.text_buffer.push(character);
+            }
+            input::Event::MouseWheel { delta_x, delta_y } => {
+                input.mouse_wheel = Point::new(delta_x, delta_y);
+            }
+            input::Event::KeyboardInput {
+                key_code,
+                state
+            } => {
+                match state {
+                    input::ButtonState::Pressed => input.pressed_keys.push(key_code),
+                    input::ButtonState::Released => input.released_keys.push(key_code),
+                };
+            }
+            input::Event::MouseInput {
+                state,
+                button
+            } => {
+                match state {
+                    input::ButtonState::Pressed => input.pressed_mouse.push(button),
+                    input::ButtonState::Released => input.released_mouse.push(button),
+                };
+            }
+            _ => {}
+        }
+    }
+
+    fn update(&mut self, _view: &Self::View, _window: &Window) {
+    }
+
+    fn interact(&mut self, input: &mut Input, _view: &mut View, _gpu: &mut Gpu) {
+        self.mouse_x = input.cursor_position.x;
+        self.mouse_y = input.cursor_position.y;
+
+        self.wheel_x = input.mouse_wheel.x;
+        self.wheel_y = input.mouse_wheel.y;
+        input.mouse_wheel.x = 0.0;
+        input.mouse_wheel.y = 0.0;
+
+        if !input.text_buffer.is_empty() {
+            for c in input.text_buffer.chars() {
+                match c {
+                    '\u{0008}' => {
+                        self.text_buffer.pop();
+                    }
+                    _ => {
+                        if self.text_buffer.chars().count() < 30 {
+                            self.text_buffer.push_str(&input.text_buffer);
+                        }
+                    }
+                }
+            }
+            input.text_buffer.clear();
+        }
+
+        self.pressed_keys = input.pressed_keys.iter()
+                                 .map(|k| format!("{:?}", k))
+                                 .collect::<Vec<_>>()
+                                 .join(", ");
+
+        self.released_keys = input.released_keys.iter()
+                                 .map(|k| format!("{:?}", k))
+                                 .collect::<Vec<_>>()
+                                 .join(", ");
+
+        input.pressed_keys.clear();
+        input.released_keys.clear();
+
+        self.pressed_mousebuttons = input.pressed_mouse.iter()
+                                 .map(|k| format!("{:?}", k))
+                                 .collect::<Vec<_>>()
+                                 .join(", ");
+
+        self.released_mousebuttons = input.released_mouse.iter()
+                                 .map(|k| format!("{:?}", k))
+                                 .collect::<Vec<_>>()
+                                 .join(", ");
+
+        input.pressed_mouse.clear();
+        input.released_mouse.clear();
+    }
```

Comment:
Right now, that the feedback flashes quite a bit and the keys disappear quite fast (I have a 144 Hz monitor, and `interact` is called every frame, so it is probably worse on my end).

What if, instead of showing pressed and release events, we just show the currently pressed keys? This would simplify the example overall, while still demonstrating the same functionality. We should be able to achieve this by using a `HashSet` for keyboard keys and another one for mouse buttons.

The `MouseWheel` event flashes too. I think we can just overwrite `input.mouse_wheel` in `on_input` and just show the value of the last scroll event, so no need to clear it in `interact`.

These simple changes should be enough. Let me know what you think!